analysis/generate_report.py

```diff
@@ -177,6 +175,11 @@ def generate_report(experiment_names,
         experiment_df = data_utils.clobber_experiments_data(
             experiment_df, experiment_names)
 
+    # Save the filtered raw data along with the report if not using cached data
+    # or if the data does not exist.
+    if not from_cached_data or not os.path.exists(data_path):
```

Comment:
The `or not os.path.exists(data_path)` results in wrong behavior.
What if I regenerate a report with different flags?

Reply:
If you are regenerating a report, you probably would be using cached data flag to reuse data. and then it wont overwrite.

This is important if you are reusing report directory. lets say you generated report with experiment 1 and then try to generate report with experiment 2, you would want to make sure that you have data.csv from experiment 2. wdyt ?